packages/astro/src/runtime/server/astro-island.ts

```diff
@@ -127,9 +127,22 @@ declare const Astro: {
 						if (!closest?.isSameNode(this)) continue;
 						slots[slot.getAttribute('name') || 'default'] = slot.innerHTML;
 					}
-					const props = this.hasAttribute('props')
-						? JSON.parse(this.getAttribute('props')!, reviver)
-						: {};
+
+					let props: Record<string, unknown>;
+
+					try {
+						props = this.hasAttribute('props')
+							? JSON.parse(this.getAttribute('props')!, reviver)
+							: {};
+					} catch (e) {
+						// eslint-disable-next-line no-console
+						console.error(
+							`[hydrate] Error parsing props for component ${this.Component?.__name}`,
```

Comment:
Not sure if `__name` will always exist, but we should also have `component-url` and `component-export` attributes to surface what (final, built) file is causing the problem.

Reply:
Cool, let me do some testing with those locally and I'll update the PR.